### ISSUE TITLE
[raft: debug]: topology_custom.test_old_ip_notification_repro failed

### ISSUE THREAD

**Issue by Annamikhlin** [scylladb/scylladb#14774]
Part of the commit https://github.com/scylladb/scylladb/commit/00db47292bdbd21e8fa72a6f681ee30e625b90bd was added new test `test_old_ip_notification_repro` - currently failing build ARM and x86 in debug mode

link to log: 
https://jenkins.scylladb.com/job/scylla-master/job/build/1539/artifact/testlog/aarch64/debug/topology_custom.test_old_ip_notification_repro.1.log
https://jenkins.scylladb.com/job/scylla-master/job/build/1539/testReport/junit/(root)/non-boost%20tests/Build__Test__Publish___buildTestPublishArm___topology_custom_test_old_ip_notification_repro_debug_1/

**Comment by Annamikhlin** [comment]
currently blocking from `master/build` to pass - @DoronArazii @mykaul FYI

**Comment by Annamikhlin** [comment]
build https://jenkins.scylladb.com/job/scylla-master/job/build/1540/testReport/ failed with the same error in `x86` and `ARM` in debug mode

**Comment by yaronkaikov** [comment]
@avikivity Maybe we should revert this commit ?

**Comment by mykaul** [comment]
Yes, especially that it tries to add a regression test for a different failure.
cc @kostja

**Comment by bhalevy** [comment]
> Let's skip debug it

Why?

https://jenkins.scylladb.com/job/scylla-master/job/build/1539/artifact/testlog/aarch64/debug/scylla-157.log
```
ERROR 2023-07-20 07:33:36,238 [shard 0] debug_error_injection - Error injection wait_for_message timeout: Condition variable timed out, at: 0xd84a9d7 /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x505cf53 /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x505ccc3 /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x505f41b /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x505fe6f /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3b4a18b /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3b49de7 0x1d90265b 0xdd2ce8b 0xdd2ca5b /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3dfbceb /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3e047c7 /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3e0833f /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3e0645b /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x386dfbb /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x386b9bb 0xd8d647f 0xe909ff3 0xe909bb7 0xe9093c7 0xdae526b 0xd8d3f9f /lib64/libc.so.6+0x2b587 /lib64/libc.so.6+0x2b65f 0xd7ea36f
   --------
   seastar::internal::coroutine_traits_base<void>::promise_type
   --------
   seastar::internal::coroutine_traits_base<void>::promise_type
   --------
   seastar::internal::coroutine_traits_base<void>::promise_type
   --------
   seastar::internal::coroutine_traits_base<void>::promise_type
   --------
   seastar::internal::coroutine_traits_base<void>::promise_type
   --------
   seastar::internal::coroutine_traits_base<void>::promise_type
   --------
   seastar::internal::coroutine_traits_base<void>::promise_type
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)>, seastar::future<void>::then_wrapped_nrvo<seastar::future<void>, seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)> >(seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)>&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)>&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)>, seastar::future<void>::then_wrapped_nrvo<seastar::future<void>, seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)> >(seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)>&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)>&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
Aborting on shard 0.
Backtrace:
  0xd84a9d7
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3fdfbe3
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3fdf7c7
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3ddad8b
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3e09dc7
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3f33f87
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3f34187
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3f34013
  linux-vdso.so.1+0x7db
  /lib64/libc.so.6+0x86c27
  /lib64/libc.so.6+0x3edff
  /lib64/libc.so.6+0x2b283
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3b49e07
  0x1d90265b
  0xdd2ce8b
  0xdd2ca5b
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3dfbceb
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3e047c7
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3e0833f
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x3e0645b
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x386dfbb
  /jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so+0x386b9bb
  0xd8d647f
  0xe909ff3
  0xe909bb7
  0xe9093c7
  0xdae526b
  0xd8d3f9f
  /lib64/libc.so.6+0x2b587
  /lib64/libc.so.6+0x2b65f
  0xd7ea36f
------ Ending test debug/topology_custom.test_old_ip_notification_repro.1::test_old_ip_notification_repro ------
```

Decoded with build_id 98ed3da16cc3d4d0aba1051f0f91e2cd0eea351e:
```
[Backtrace #0]
__interceptor_backtrace.part.0 at ??:?
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3fdfbe3 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3fdf7c7 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3ddad8b 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3e09dc7 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3f33f87 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3f34187 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3f34013 
linux-vdso.so.1 0x7db 
__pthread_setschedparam at :?
__GI___qsort_r at :?
__gconv_transform_ucs4_internal at :?
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3b49e07 
seastar::future<void> utils::error_injection<true>::injection_handler::wait_for_message<std::chrono::_V2::steady_clock, std::chrono::duration<long, std::ratio<1l, 1000000000l> > >(std::chrono::time_point<std::chrono::_V2::steady_clock, std::chrono::duration<long, std::ratio<1l, 1000000000l> > >) at ././utils/error_injection.hh:155
std::__n4861::coroutine_handle<seastar::internal::coroutine_traits_base<void>::promise_type>::resume() const at /usr/bin/../lib/gcc/aarch64-redhat-linux/12/../../../../include/c++/12/coroutine:244
seastar::internal::coroutine_traits_base<void>::promise_type::run_and_dispose() at ././seastar/include/seastar/core/coroutine.hh:125
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3dfbceb 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3e047c7 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3e0833f 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x3e0645b 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x386dfbb 
/jenkins/workspace/scylla-master/build/scylla/build/debug/seastar/libseastar.so 0x386b9bb 
scylla_main(int, char**) at ./main.cc:627
int std::__invoke_impl<int, int (*&)(int, char**), int, char**>(std::__invoke_other, int (*&)(int, char**), int&&, char**&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/12/../../../../include/c++/12/bits/invoke.h:61
std::enable_if<is_invocable_r_v<int, int (*&)(int, char**), int, char**>, int>::type std::__invoke_r<int, int (*&)(int, char**), int, char**>(int (*&)(int, char**), int&&, char**&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/12/../../../../include/c++/12/bits/invoke.h:114
std::_Function_handler<int (int, char**), int (*)(int, char**)>::_M_invoke(std::_Any_data const&, int&&, char**&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/12/../../../../include/c++/12/bits/std_function.h:290
std::function<int (int, char**)>::operator()(int, char**) const at /usr/bin/../lib/gcc/aarch64-redhat-linux/12/../../../../include/c++/12/bits/std_function.h:591
main at ./main.cc:1968
__gconv_transform_internal_ucs4le at :?
__gconv_transform_internal_ucs4le at :?
_start at ??:?
```

**Comment by bhalevy** [comment]
Same crash seen in https://jenkins.scylladb.com/view/master/job/scylla-master/job/build/1541/artifact/testlog/aarch64/debug/scylla-152.log

**Comment by bhalevy** [comment]
For reference, since this happened on aarch64/debug it could be related to https://github.com/scylladb/scylladb/issues/14412

**Comment by margdoc** [comment]
The first node (the one which crashed) works as expected. The issue is with the second node, which stops and restarts, taking more than a minute (exceeding the timeout set on the first node). During Jenkins' debug mode, it takes more time than we consider safe for the timeout.

We can either increase this timeout or skip this test in the debug mode (which was our original intention, but I forgot to update the `suite.yaml` file while changing the test's name).

I agree with Kostja to skip this test.

**Comment by kbr-scylla** [comment]
The test was supposed to be skipped in debug, let's fix that. @margdoc

**Comment by bhalevy** [comment]
I've just hit this locally in dev mode

**Comment by kbr-scylla** [comment]
Fixed by  #14777
